gui/browsable/src/RFieldProvider.hxx

```diff
@@ -38,132 +42,145 @@ using namespace std::string_literals;
 */
 
 class RFieldProvider : public RProvider {
-public:
+   class RDrawVisitor : public ROOT::Experimental::Detail::RFieldVisitor {
+   private:
+      std::shared_ptr<ROOT::Experimental::Detail::RPageSource> fNtplSource;
+      std::unique_ptr<TH1> fHist;
+
+      /** Test collected entries if it looks like integer values and one can use better binning */
+      void TestHistBuffer()
+      {
+         auto len = fHist->GetBufferLength();
+         auto buf = fHist->GetBuffer();
+
+         if (!buf || (len < 5))
+            return;
+
+         Double_t min = buf[1], max = buf[1];
+         Bool_t is_integer = kTRUE;
+
+         for (Int_t n = 0; n < len; ++n) {
+            Double_t v = buf[2 + 2*n];
+            if (v > max) max = v;
+            if (v < min) min = v;
+            if (TMath::Abs(v - TMath::Nint(v)) > 1e-5) { is_integer = kFALSE; break; }
+         }
 
-   // virtual ~RFieldProvider() = default;
+         // special case when only integer values in short range - better binning
+         if (is_integer && (max-min < 100)) {
+            max += 2;
+            if (min > 1) min -= 2;
+            int npoints = TMath::Nint(max - min);
+            fHist = std::make_unique<TH1F>(fHist->GetName(), fHist->GetTitle(), npoints, min, max);
+            fHist->SetDirectory(nullptr);
```

Comment:
Do I understand correctly that here `fHist` recreated again, providing new `min` and `max` arguments?
If yes - code is wrong. While buffer `fHist->GetBuffer()` will be destroyed together with original histogram.

Reply:
Thanks! Should be fixed in [3c02346](https://github.com/root-project/root/pull/8545/commits/3c0234621adba4b020e9d2d110c2c690c3391d38)